packages/tracing-plugin/src/tracer.ts

```diff
@@ -53,6 +56,25 @@ export default function tracer(
   const moduleMap = _opts?.moduleMap || {}
   const moduleMapKeys = Object.keys(moduleMap).map((k) => [k, new RegExp(k)] as const)
 
+  const programDir = _program.getCurrentDirectory()
+  let internalPackage = false
+  const packageJsonDir = findUp(programDir, "package.json")
+
+  if (packageJsonDir) {
+    try {
+      const packageJsonContent = fs.readFileSync(
+        path.join(packageJsonDir, "package.json"),
+        "utf-8"
+      )
+      const obj = JSON.parse(packageJsonContent)
+      if (typeof obj === "object" && INTERNAL_PACKAGES.includes(obj.name)) {
```

Comment:
Potentially, name could be a false positive match for some private packages. We could potentially, instead of checking for the name of the package, see if there is a config field set to specify that it is an effect internal package. It would be kinda of "hidden feature" - that is if someone set's that option in package.json, they will shoot themselves in a foot and break their own traces.

Reply:
I don't get this, there are only 3 cases:
1) system => needs to be local paths
2) core => needs to reference system
3) any other need to reference core

Reply:
That's what I thought too, but I had a bug with tests.

In short, tests run from top level folder, where the package name is `'effect-ts'`, so the import that's being used in tests is `@effect-ts/core/Tracing`. The paths in tsconfig.json would reference built files rather than real sources, so then there would actually be 2 different instances of `Tracing/index.ts` files being used, and so one trace was being lost (the one in tuple test cases). I'm guessing that since tests are now linked against real source, we can actually return to this behaviour and resolve to `@effect-ts/core/Tracing` even for tests and it should work. I could try that.

Reply:
I changed it, so now in system it should be all relative paths. I also removed exception in tests, so it now links against `@effect-ts/core`. This seems to work locally. But it means that in jest configuration we have to link against `src` rather than `build`, which is good in my opinion anyway.